core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java

```diff
@@ -81,6 +81,10 @@
             "LEFT JOIN FETCH ig.securityGroup WHERE s.id= :id AND (s.type is not 'TEMPLATE' OR s.type is null)")
     Optional<Stack> findOneWithLists(@Param("id") Long id);
 
+    @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.resources LEFT JOIN FETCH s.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData "
+            + "LEFT JOIN FETCH ig.template LEFT JOIN FETCH ig.securityGroup WHERE s.id= :id AND (s.type is not 'TEMPLATE' OR s.type is null)")
```

Comment:
all these extra stuff is really necessary? I expected to see only one join for the resources here

Reply:
Yes, there is still functionality that requires the joins unfrotunatley.